### PR TITLE
sourcery-version-check.inc: ignore the minor version

### PR DESCRIPTION
**Pull request by kergoth**
We don't need to abort the build if we have 12.0.4 but it wants 12.0.0.

JIRA: SB-16597

### CODE REVIEW

**Comment by Noor-Ahsan** [comment]
what will be the result if flex is release with 2.0.5 toolchain version anf down the road we get 2.0.20

**Comment by kergoth** [comment]
[**@Noor-Ahsan**](https://github.com/Noor-Ahsan) commented on [Mar 17, 2021, 11:08 AM MST](https://github.com/MentorEmbedded/meta-mentor/pull/1404#issuecomment-801299116 "2021-03-17T18:08:34Z - Replied by Github Reply Comments"):
> what will be the result if flex is release with 2.0.5 toolchain version anf down the road we get 2.0.20

Then it'd allow it to be used. You can't have it both ways, either we check minor versions or we don't.

**Comment by Noor-Ahsan** [comment]
> [**@Noor-Ahsan**](https://github.com/Noor-Ahsan) commented on [Mar 17, 2021, 11:08 AM MST](https://github.com/MentorEmbedded/meta-mentor/pull/1404#issuecomment-801299116):
> 
> > what will be the result if flex is release with 2.0.5 toolchain version anf down the road we get 2.0.20
> 
> Then it'd allow it to be used. You can't have it both ways, either we check minor versions or we don't. That said, I doubt most customers are trying to use a mel with a mismatched toolchain version to begin with.

in case of 2.0.20 ver.pop will pop 20 or just 0? that I was not sure.if it pops 20 then we are good. two ver.pop will pop 20 and .

**Comment by kergoth** [comment]
[**@Noor-Ahsan**](https://github.com/Noor-Ahsan) commented on [Mar 17, 2021, 11:40 AM MST](https://github.com/MentorEmbedded/meta-mentor/pull/1404#issuecomment-801321497 "2021-03-17T18:40:13Z - Replied by Github Reply Comments"):
> > [**@Noor-Ahsan**](https://github.com/Noor-Ahsan) commented on [Mar 17, 2021, 11:08 AM MST](https://github.com/MentorEmbedded/meta-mentor/pull/1404#issuecomment-801299116):
> > 
> > > what will be the result if flex is release with 2.0.5 toolchain version anf down the road we get 2.0.20
> > 
> > Then it'd allow it to be used. You can't have it both ways, either we check minor versions or we don't. That said, I doubt most customers are trying to use a mel with a mismatched toolchain version to begin with.
> 
> in case of 2.0.20 ver.pop will pop 20 or just 0? that I was not sure.if it pops 20 then we are good. two ver.pop will pop 20 and .

Yeah, as the comment indicates, it pops the 20 and the period before it, the separators are kept when explode_version splits it up.

**Comment by Noor-Ahsan** [comment]
Can we update this message 
bb.fatal('Found toolchain version `%s`, expected `%s`. Please install the supported toolchain.' % ('-'.join(sourcery_version), '-'.join(toolchain_required_version)))

as this is using stripped variables

File: meta-mel/conf/distro/include/sourcery-version-check.inc [hunk 1]
```diff
@@ -10,13 +10,15 @@ python sourcery_version_check () {
         if str is None:
             return []
 
-        elements = str.split('-')
-        if len(elements) > 1:
-            elements.pop()
-        return elements
+        ver = bb.utils.explode_version(str)
+        if len(ver) > 4:
```
Comment (abelal) [inline meta-mel/conf/distro/include/sourcery-version-check.inc:14]: Shouldn't this be a while?

**Comment by kergoth** [comment]
@Noor-Ahsan @abelal Reworked to simplify and adjust the message.

**Review by Noor-Ahsan — APPROVED** [review]

**Review by abelal — APPROVED** [review]
Looks fine!

**Pull request merged**